src/libxively/sft/xi_sft_logic_file_chunk_handlers.c

```diff
@@ -40,10 +40,10 @@ xi_sft_on_message_file_chunk_process_file_chunk( xi_sft_context_t* context,
     /* write bytes through FILE BSP */
     size_t bytes_written = 0;
 
-    state =
+    state = xi_fs_bsp_io_fs_2_xi_state( 
```

Comment:
So, again, why do we do this conversion? Why isn't BSP IO FS allowed to return xi_state_t? The xively_error.h is already accessible, it is on the API? Doesn't this sound like an unnecessary overhead?

Reply:
See my main reply comment above.